README.md

```diff
@@ -8,4 +8,249 @@ are in the process of being changed and will certainly break your code.
+To create/retrieve a key the user calls:
+
+Create/retrieve key:
+
+    if err, key1 := tags.CreateKeyString("keyNameID1"); err != nil {
+        // handle error
+    }
+    ...
+
+    if err, key2 := tags.CreateKeyString("keyNameID2"); err != nil {
+        // handle error
+    }
+    ...
+
+### Create a set of tags associated with keys
+To create a new tag set from scratch using changes:
+
+    tsb := &tags.TagSetBuilder{}
+    tsb..StartFromEmpty()
+    tsb.InsertString( key1, "foo value")
+    tsb.UpdateString( key1, "foo value2")
+    tsb.UpsertString( key2, "bar value")
+    tagsSet := tsb.Build()
+
+    // A shorter way of achieving the same is:
+    tagsSet := &tags.TagSetBuilder{}.StartFromEmpty()
+                                    .InsertString( key1, "foo value")
+                                    .UpdateString( key1, "foo value2")
+                                    .UpsertString( key2, "bar value")
+                                    .Build()
+
+
+
+To create a new tagsSet from an existing tag set oldTagSet:
+    oldTagSet := ...
+    tsb := &tags.TagSetBuilder{}
+    tsb.StartFromTagSet(oldTagSet)
+    tsb.InsertString( key1, "foo value")
+    tsb.UpdateString( key1, "foo value2")
+    tsb.UpsertString( key2, "bar value")
+    newTagSet := tsb.Build()
+
+    // A shorter way of achieving the same is:
+    oldTagSet := ...
+    newTagSet := &tags.TagSetBuilder{}.StartFromTagSet(oldTagSet)
+                                      .InsertString( key1, "foo value")
+                                      .UpdateString( key1, "foo value2")
+                                      .UpsertString( key2, "bar value")
+                                      .Build()
+
+### Add new tagSet to a context / Modify tagSet in a context 
+Add tags to a context for propagation to downstream methods and downstream rpcs:
+To create a new context with the tags. This will create a new context where all the existing tags in the current context are deleted and replaced with the tags passed as argument.    
+    
+    newTagSet  := ...
+    ctx2 := tags.ContextWithNewTags(ctx, newTagSet)
+
+Create a new context keeping the old tag set and adding new tags to it, removing specific tags from it, or modifying the values fo some tags.
+    
+    oldTagSet := tags.FromContext(ctx)
+    newTagSet := &tags.TagSetBuilder{}.StartFromTagSet(oldTagSet)
+                                      .InsertString( key1, "foo value")
+                                      .UpdateString( key1, "foo value2")
+                                      .UpsertString( key2, "bar value")
+                                      .Build()
+    ctx2 := tags.ContextWithChanges(ctx, newTagSet)
+
+## Registering views and retrieving their collected data.
+
+### To create/register a measure a.k.a resource
+Create/load measures units:
+
+    // returns a *MeasureFloat64
+    mf := stats.NewMeasureFloat64("/my/float64/measureName", "some measure")
```

Comment:
Should we have error check here?

Reply:
The error is to be checked during registration.